perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -1870,6 +1874,32 @@ sub ban_user : Private {
     return 1;
 }
 
+sub unban_user : Private {
+    my ( $self, $c, $user ) = @_;
+
+    if ($user->email_verified && $user->email) {
+        my $abuse = $c->model('DB::Abuse')->find({ email => $user->email });
+        if ( $abuse ) {
+            $abuse->delete;
+            $c->stash->{status_message} = _('user removed from abuse list');
+        } else {
+            $c->stash->{status_message} = _('user not in abuse list');
+        }
+        $c->stash->{username_in_abuse} = 0;
+    }
+    if ($user->phone_verified && $user->phone) {
+        my $abuse = $c->model('DB::Abuse')->find({ email => $user->phone });
+        if ( $abuse ) {
+            $abuse->delete;
+            $c->stash->{status_message} = _('user removed from abuse list');
+        } else {
+            $c->stash->{status_message} = _('user not in abuse list');
```

Comment:
If they have both verified email and phone, and email was in abuse and phone wasn't, the message output will be "user not in abuse list" instead of "user removed". Given the button is only shown if they're in the list, probably can just set status_message if something is removed, and not have a message in the case it isn't.

Reply:
You could also probably combine these two parts with a search on `email` being either email or phone (if verified), then calling delete on that resultset?